cmdfactory/help.go

```diff
@@ -268,3 +269,20 @@ func dedent(s string) string {
 	}
 	return strings.TrimSuffix(buf.String(), "\n")
 }
+
+// returns whether the given list contains at least one of the provided
+// elements.
+func containsOneOf(inputs []string, elems ...string) bool {
+	elemsSet := set.NewStringSet()
+	for _, elem := range elems {
+		elemsSet.Add(elem)
+	}
+
+	for _, input := range inputs {
+		if elemsSet.Contains(input) {
+			return true
+		}
+	}
+
+	return false
+}
```

Comment:
This utility function can be repurposed and imho should be placed within the `internal/set` package as `ContainsOneOf` and part of `stringSet`.

The [`NewStringSet()`](https://github.com/unikraft/kraftkit/blob/staging/internal/set/string_set.go#L33-L38) method's prototype should also be updated to `NewStringSet(base ...string)` for faster use of `AddValues`.

Reply:
Thus:
```diff
-	if isRootCmd(cmd.Parent()) && len(args) >= 2 && !containsOneOf(args, "--help", "-h") {
+	argSet := set.NewStringSet(args...)
```

Reply:
I was thinking exactly the same, the current set implementation is lacking a proper initializer.
Pushing as a separate PR.

Reply:
If you are creating a separate PR, please can you include a replacement (and deprecation of) the more primitive [`utils.Contains`](https://github.com/unikraft/kraftkit/blob/staging/utils/utils.go#L117-L125) method which is used more widely in the project. There is currently little-to-no use of `stringSet` which is more specific and has its own package. The `utils` package is anyway a bit of a odd package and left over from a some verbatim imports a long time ago. Using the `set` package is preferable.

Reply:
String set changes are in #439
I'm addressing the change to the utils package now in its own PR.